### PR TITLE
Resolve a few more warnings, potential bugs

### PR DESCRIPTION
**Pull request by adizaimi**
Resolve a few more warnings, potential bugs

### CODE REVIEW

File: berkdb/db/db_meta.c [hunk 1]
```diff
@@ -128,21 +128,20 @@ __db_new_from_freelist(DBC *dbc, DBMETA *meta, u_int32_t type, PAGE **pagepp)
...
-		 * We want to take the first page off the free list and
-		 * then set meta->free to the that page's next_pgno, but
-		 * we need to log the change first.
-		 */
-		newnext = h->next_pgno;
-		lsn = h->lsn;
-		extend = 0;
 	}
 
+    pgno = meta->free;
```
Comment (riverszhang89) [inline berkdb/db/db_meta.c]: tabs

File: tools/cdb2sockpool/cdb2sockpool.c [hunk 2]
```diff
@@ -220,11 +220,11 @@ int recvall(int fd, void *bufp, int len)
 
     while (bytesleft) {
         rc = recv(fd, buf + off, bytesleft, MSG_WAITALL);
-        if (rc == -1) {
-            if (rc == EINTR || rc == EAGAIN)
-                continue;
+        if (rc == EINTR || rc == EAGAIN)
```
Comment (mponomar) [inline tools/cdb2sockpool/cdb2sockpool.c]: I think this should be `if (rc ==-1 && errno == EINTR || errno == EAGAIN)`

Comment (adizaimi) [reply tools/cdb2sockpool/cdb2sockpool.c]: yes, thanks. Fixed.

File: berkdb/db/db_meta.c (hunk 1 above)
Comment (adizaimi) [reply berkdb/db/db_meta.c]: yep. thanks.

**Pull request merged**